app/models/recipe.rb

```diff
@@ -1,3 +1,10 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost 
+      Recipe.select("SUM(ingredients.cost) AS total").joins(recipe_ingredients: :ingredient)[0].total
```

Comment:
A sufficient query here would be `ingredients.sum(:cost)` - I think you over-engineered it a bit with this join. Keep in mind that we are getting this total for a *specific instance* of a recipe, so we can use the associated ingredients rather than needing to join to the ingredients table.